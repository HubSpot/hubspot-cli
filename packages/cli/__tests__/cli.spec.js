const path = require('path');
const cmd = require('../lib/cmd');
const rimraf = require('rimraf');
const { existsSync, readFileSync } = require('fs');
const yaml = require('js-yaml');

describe('my cli program', () => {
  const cliPath = path.join(__dirname, '../bin/cli.js');
  const cliProcess = cmd.create(cliPath, '.');

  it('should print the correct output', async () => {
    rimraf.sync('./hubspot.config.yml');

    await cliProcess.execute(
      ['init', '--qa'], // args
      [
        cmd.ENTER,
        'CiRjYTYyMmY0YS0xODQyLTQyMmYtOTc3OC1hMDEzMDExZTRjMjMQ8tW5MRin-NoBKhkABeaRgpKG4ptdbYYuc5IuNjUG7DE4CE0F',
        cmd.ENTER,
        'QA',
        cmd.ENTER,
      ],
      { env: { DEBUG: false }, timeout: 500 }
    );

    expect(existsSync('./hubspot.config.yml')).toBe(true);

    expect(
      yaml.safeLoad(readFileSync('./hubspot.config.yml', 'utf8'))
    ).toMatchSnapshot({
      portals: [{ auth: { tokenInfo: expect.any(Object) } }],
    });
  });
});
