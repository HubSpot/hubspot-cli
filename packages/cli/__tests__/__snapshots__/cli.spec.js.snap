// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my cli program should print the correct output 1`] = `
Object {
  "defaultPortal": "QA",
  "portals": Array [
    Object {
      "auth": Object {
        "tokenInfo": Any<Object>,
      },
      "authType": "personalaccesskey",
      "env": "qa",
      "name": "QA",
      "personalAccessKey": "CiRjYTYyMmY0YS0xODQyLTQyMmYtOTc3OC1hMDEzMDExZTRjMjMQ8tW5MRin-NoBKhkABeaRgpKG4ptdbYYuc5IuNjUG7DE4CE0F",
      "portalId": 103705330,
    },
  ],
}
`;
